python/ic_azure/insights_metric.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+
+# Python imports
+from datetime import datetime, timedelta
+from argparse import ArgumentParser
+
+# Azure client imports
+from azure_client import AzureClient
+
+
+class InsightsMetric(object):
+    """Retrieve metric data from Azure's Microsoft Insights components."""
+
+    def __init__(self, azure_client):
+        self._client = azure_client.client()
+        self.subscription_id = azure_client.subscription_id
+
+    # Method to retrieve metrics from Microsoft Insight resources
+    def get_metric(self, resource_group, resource_name, interval, metric,
+                   statistic, timeshift):
+
+        # Declare variables
+        ret_val = -1
+
+        # Calculate timestamps
+        end_time = datetime.utcnow() - timedelta(seconds=timeshift)
+        start_time = end_time - timedelta(seconds=interval)
+
+        # Create resource ID
+        resource_id = "subscriptions/{}/resourceGroups/{}".format(
+            self.subscription_id,
+            resource_group
+        )
+        resource_id += "/providers/microsoft.insights/components/{}".format(
+            resource_name
+        )
+
+        # Get timegrain based on interval
+        # Possible values: None, PT1M, PT1H, P1D
+        quotient, remainder = divmod(interval, 60)
+        if quotient <= 1:
+            timegrain = 'PT1M'
+        elif quotient <= 60:
+            timegrain = 'PT1H'
+        else:
+            timegrain = 'P1D'
+
+        # Retrieve metric data
+        metrics_data = self._client.metrics.list(
+            resource_id,
+            timespan="{}/{}".format(
+                start_time.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                end_time.strftime('%Y-%m-%dT%H:%M:%SZ')
+            ),
+            interval=timegrain,
+            metricnames=metric,
+            aggregation=statistic
+        )
+
+        # Loop through metric data and retrieve relevant value
+        for item in metrics_data.value:
+            for timeserie in item.timeseries:
+                for data in timeserie.data:
+                    ret_val = data.total
```

Comment:
I assume the data is in ascending timestamp order so this loop picks the latest value in case the query would return multiple values? I assume that the logic above tries to ensure that the query always returns just one value?

Reply:
Exactly. Until we resolve the issue with using the filters, user might retrieve multiple values which I tought is something we do not want.